app/uk/gov/hmrc/agentmappingfrontend/audit/AuditService.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentmappingfrontend.audit
+
+import javax.inject.{Inject, Singleton}
+
+import play.api.mvc.Request
+import uk.gov.hmrc.agentmappingfrontend.audit.AgentFrontendMappingEvent.AgentFrontendMappingEvent
+import uk.gov.hmrc.domain.SaAgentReference
+import uk.gov.hmrc.play.audit.AuditExtensions.auditHeaderCarrier
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.play.audit.model.DataEvent
+import uk.gov.hmrc.play.http.HeaderCarrier
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.Try
+
+object AuditService {
+
+  def auditCheckAgentRefCodeEvent(saAgentReferenceOpt: Option[SaAgentReference], authProviderId: Option[String], authProviderType: Option[String])
+                                 (auditService: AuditService)
+                                 (implicit hc: HeaderCarrier, request: Request[Any]): Unit = {
+    val event = createEvent(AgentFrontendMappingEvent.CheckAgentRefCode,
+      Seq("isEnrolledSAAgent" -> saAgentReferenceOpt.isDefined)
+        ++ saAgentReferenceOpt.map(r => Seq("saAgentRef" -> r.value)).getOrElse(Seq.empty)
+        ++ authProviderId.map(v => Seq("authProviderId" -> v)).getOrElse(Seq.empty)
+        ++ authProviderType.map(v => Seq("authProviderType" -> v)).getOrElse(Seq.empty)
+    )
+    auditService.send(event)
+  }
+
+  private def createEvent(event: AgentFrontendMappingEvent,
+                          details: Seq[(String, Any)])
+                         (implicit hc: HeaderCarrier, request: Request[Any]): DataEvent = {
+    DataEvent(
+      auditSource = "agent-mapping-frontend",
+      auditType = event.toString,
+      tags = hc.toAuditTags("", request.path),
```

Comment:
I think we should probably supply a transactionName (instead of "") - for subscription I just used "Agent services subscription" but I suggest asking Graeme Blackwood what value we should use (you can ask on the JIRA or in #event-agent-auditing)